### ISSUE TITLE
WrapperPotentials don't work with Orbits class

### ISSUE THREAD

**Issue by jmackereth** [jobovy/galpy#367]
I'm trying to use the new `Orbits` class to integrate many orbits inside a wrapper potential. The orbits seem to integrate (very fast now thanks to this great addition!) but im coming up against the following error (i've pasted the full error output below) when I try to access the orbits:
```python
ValueError: WrapperPotentials are only supported in 3D and 2D
```

It looks like something to do with copying the orbit instances, but I cant seem to figure out what's going on... Integrating just one orbit in the usual way still works, of course.

Perhaps this is related to issue #360 ?

more detail:
You should be able to recreate the error like this:
```python
import numpy as np
from galpy.orbit import Orbit, Orbits
from galpy.potential import MWPotential2014
from galpy.potential import SolidBodyRotationWrapperPotential, DehnenSmoothWrapperPotential, \
   CosmphiDiskPotential, DehnenBarPotential, LogarithmicHaloPotential, SpiralArmsPotential

#initialise a wrapper potential
tform= -10.
tsteady= 5. 
omega= 1.85 
angle=25./180.*np.pi 
dp= DehnenBarPotential(omegab=omega,rb=3.5/8.,Af=(1./75.),tform=tform,tsteady=tsteady,barphi=angle)
lhp=LogarithmicHaloPotential(normalize=1.)
dswp= DehnenSmoothWrapperPotential(pot=dp,tform=-4.*2.*np.pi/dp.OmegaP(),tsteady=2.*2.*np.pi/dp.OmegaP())
pot= [lhp,dswp]

#initialise 100 random orbits
r = np.random.randn(100)*0.01+1.
z = np.random.randn(100)*0.01+0.2
phi = np.random.randn(100)*0.01+0.
vR = np.random.randn(100)*0.01+0.
vT = np.random.randn(100)*0.01+1.
vz = np.random.randn(100)*0.01+0.02
vxvv = np.dstack([r,vR,vT,z,vz,phi])[0]
os = Orbits(vxvv)

times = np.linspace(0.,100.,3000)
os.integrate(times, pot)
```

The code above runs, then if we try to get the first orbit:

```python
os[0].getOrbit()
```
we get these errors:
```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-120-fbe7bd7475c2> in <module>()
----> 1 os[0].plot()

~/anaconda3/lib/python3.6/site-packages/galpy-1.5.dev0-py3.6-linux-x86_64.egg/galpy/orbit/Orbits.py in __getitem__(self, key)
    253             if key < 0 : # negative indices
    254                 key+= len(self)
--> 255             return copy.deepcopy(self._orbits[key])
    256         elif isinstance(key,slice):
    257             orbits_list= [copy.deepcopy(self._orbits[ii]) 

~/anaconda3/lib/python3.6/copy.py in deepcopy(x, memo, _nil)
    178                     y = x
    179                 else:
--> 180                     y = _reconstruct(x, memo, *rv)
    181 
    182     # If is its own copy, don't memoize.

~/anaconda3/lib/python3.6/copy.py in _reconstruct(x, memo, func, args, state, listiter, dictiter, deepcopy)
    278     if state is not None:
    279         if deep:
--> 280             state = deepcopy(state, memo)
    281         if hasattr(y, '__setstate__'):
    282             y.__setstate__(state)

~/anaconda3/lib/python3.6/copy.py in deepcopy(x, memo, _nil)
    148     copier = _deepcopy_dispatch.get(cls)
    149     if copier:
--> 150         y = copier(x, memo)
    151     else:
    152         try:

~/anaconda3/lib/python3.6/copy.py in _deepcopy_dict(x, memo, deepcopy)
    238     memo[id(x)] = y
    239     for key, value in x.items():
--> 240         y[deepcopy(key, memo)] = deepcopy(value, memo)
    241     return y
    242 d[dict] = _deepcopy_dict

~/anaconda3/lib/python3.6/copy.py in deepcopy(x, memo, _nil)
    178                     y = x
    179                 else:
--> 180                     y = _reconstruct(x, memo, *rv)
    181 
    182     # If is its own copy, don't memoize.

~/anaconda3/lib/python3.6/copy.py in _reconstruct(x, memo, func, args, state, listiter, dictiter, deepcopy)
    278     if state is not None:
    279         if deep:
--> 280             state = deepcopy(state, memo)
    281         if hasattr(y, '__setstate__'):
    282             y.__setstate__(state)

~/anaconda3/lib/python3.6/copy.py in deepcopy(x, memo, _nil)
    148     copier = _deepcopy_dispatch.get(cls)
    149     if copier:
--> 150         y = copier(x, memo)
    151     else:
    152         try:

~/anaconda3/lib/python3.6/copy.py in _deepcopy_dict(x, memo, deepcopy)
    238     memo[id(x)] = y
    239     for key, value in x.items():
--> 240         y[deepcopy(key, memo)] = deepcopy(value, memo)
    241     return y
    242 d[dict] = _deepcopy_dict

~/anaconda3/lib/python3.6/copy.py in deepcopy(x, memo, _nil)
    148     copier = _deepcopy_dispatch.get(cls)
    149     if copier:
--> 150         y = copier(x, memo)
    151     else:
    152         try:

~/anaconda3/lib/python3.6/copy.py in _deepcopy_list(x, memo, deepcopy)
    213     append = y.append
    214     for a in x:
--> 215         append(deepcopy(a, memo))
    216     return y
    217 d[list] = _deepcopy_list

~/anaconda3/lib/python3.6/copy.py in deepcopy(x, memo, _nil)
    178                     y = x
    179                 else:
--> 180                     y = _reconstruct(x, memo, *rv)
    181 
    182     # If is its own copy, don't memoize.

~/anaconda3/lib/python3.6/copy.py in _reconstruct(x, memo, func, args, state, listiter, dictiter, deepcopy)
    272     if deep and args:
    273         args = (deepcopy(arg, memo) for arg in args)
--> 274     y = func(*args)
    275     if deep:
    276         memo[id(x)] = y

~/anaconda3/lib/python3.6/copyreg.py in __newobj__(cls, *args)
     86 
     87 def __newobj__(cls, *args):
---> 88     return cls.__new__(cls, *args)
     89 
     90 def __newobj_ex__(cls, args, kwargs):

~/anaconda3/lib/python3.6/site-packages/galpy-1.5.dev0-py3.6-linux-x86_64.egg/galpy/potential/WrapperPotential.py in __new__(cls, *args, **kwargs)
     24             parentWrapperPotential= WrapperPotential
     25         else:
---> 26             raise ValueError("WrapperPotentials are only supported in 3D and 2D")
     27         # Create object from custom class that derives from correct wrapper,
     28         # make sure to turn off normalization for all wrappers

ValueError: WrapperPotentials are only supported in 3D and 2D
```

**Comment by jobovy** [comment]
Thanks for the very clear report, I can reproduce this. Yes, seems like the issue is with copying the potential instance when the orbit is copied, which seems to work for regular potentials, but not for WrapperPotentials. This could very well be related to #360; @mwbub will put in a fix for #360 over the weekend and we can see whether that fixes the problem and revisit if it doesn't.

**Comment by jobovy** [comment]
@mwbub's fix to #360 (merged into ``orbits``) does indeed seem to have fixed this issue as well, but please test.

**Comment by jmackereth** [comment]
Looks like this is definitely fixed. Really great work!
